pandas/core/internals/concat.py

```diff
@@ -283,8 +289,18 @@ def _concat_managers_axis0(
             nb._mgr_locs = nb._mgr_locs.add(offset)
             blocks.append(nb)
 
+        if not made_copy and not copy and using_copy_on_write():
+            refs.extend([weakref.ref(blk) for blk in mgr.blocks])
+            parents.append(mgr)
+
         offset += len(mgr.items)
-    return BlockManager(tuple(blocks), axes)
+
+    result_parents = parents if parents else None
+    result_ref = refs if refs else None
+    result = BlockManager(tuple(blocks), axes, parent=result_parents, refs=result_ref)
+    result.parent = result_parents
+    result.refs = result_ref
```

Comment:
```suggestion
    result = BlockManager(tuple(blocks), axes, parent=result_parents, refs=result_ref)
```

Reply:
Yikes, thx. not sure what I was thinking here